src/error.rs

```diff
@@ -1,26 +1,65 @@
+use thiserror::Error;
+
+use crate::proto::ConnectError;
+
+/// The set of observable errors when interacting with a Faktory server.
+#[derive(Debug, Error)]
+#[allow(clippy::manual_non_exhaustive)]
+pub enum Error {
+    /// The connection to the server, or one of its prerequisites, failed.
+    #[error("connection error: {0}")]
```

Comment:
I think the recommendation is to _either_ return inner errors through `Error::source` (which `#[from]` does) _or_ print the inner error in `Display`, but never both (see https://github.com/rust-lang/project-error-handling/issues/27#issuecomment-763950178, https://github.com/rust-lang/project-error-handling/issues/23, https://github.com/rust-lang/api-guidelines/pull/210). So I think these errors shouldn't include `: {0}` where `.0` is an inner error.